### PR TITLE
Bug 1749293 - Apply debug features before sending pings at init

### PR DESCRIPTION
**Pull request by brizental**
### Pull Request checklist ###

- [x] **Quality**: Make sure this PR builds and runs cleanly.
  - Inside the `glean/` folder, run:
    - `npm run test` Runs _all_ tests
    - `npm run lint` Runs _all_ linters
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes a changelog entry to `CHANGELOG.md` or an explanation of why it does not need one
- [ ] ~**Documentation**: This PR includes documentation changes, an explanation of why it does not need that or a follow-up bug has been filed to do that work~

### CODE REVIEW

File: glean/src/core/glean.ts [hunk 1]
```diff
@@ -37,6 +37,13 @@ namespace Glean {
   // An instance of the ping uploader.
   export let pingUploader: PingUploadManager;
 
+  // Temporary holders for debug values,
+  // to be used when these values are set before initialize
+  // and can be applied during initialized.
+  export let debugViewTag: string | undefined;
+  export let logPings: boolean | undefined;
+  export let sourceTags: string[] | undefined;
```
Comment (chutten) [inline glean/src/core/glean.ts:45]: Would it be worth prefixing these members with `preinit` or something to make clear their expected lifetime of usefulness? cf. https://searchfox.org/glean/search?path=&q=PRE_INIT

File: glean/src/index/base.ts [hunk 2]
```diff
@@ -105,9 +103,7 @@ export default (platform: Platform): {
      *
      * Ping tags will show in the destination datasets, after ingestion.
      *
-     * Note** Setting `sourceTags` will override all previously set tags.
-     *
-     * To unset the `sourceTags` call `Glean.unsetSourceTags();
+     * Note Setting `sourceTags` will override all previously set tags.
```
Comment (chutten) [inline glean/src/index/base.ts:106]:
```suggestion
     * Note: Setting `sourceTags` will override all previously set tags.
```

File: glean/tests/unit/core/glean.spec.ts [hunk 3]
```diff
@@ -430,6 +430,23 @@ describe("Glean", function() {
     assert.strictEqual(Context.config.sourceTags, undefined);
   });
 
+  it("enabling debug features before init apply to ping sent at initialize", async function() {
+    await testUninitializeGlean(false);
+
+    Glean.setLogPings(true);
+    Glean.setDebugViewTag("test");
+    Glean.setSourceTags(["hey", "ho"]);
```
Comment (chutten) [inline glean/tests/unit/core/glean.spec.ts:438]: let's go

File: glean/tests/unit/core/glean.spec.ts [hunk 4]
```diff
@@ -430,6 +430,23 @@ describe("Glean", function() {
     assert.strictEqual(Context.config.sourceTags, undefined);
   });
 
+  it("enabling debug features before init apply to ping sent at initialize", async function() {
+    await testUninitializeGlean(false);
+
+    Glean.setLogPings(true);
```
Comment (chutten) [inline glean/tests/unit/core/glean.spec.ts:436]: I guess the test only tests that this doesn't explode, we can't inspect the log?

**Review by chutten — APPROVED** [review]
Some naming and other nits, but could land without any adjustments if you deem none are worth applying.

File: glean/tests/unit/core/glean.spec.ts (hunk 4 above)
Comment (brizental) [reply glean/tests/unit/core/glean.spec.ts:436]: Yeah we can inspect the log... It's just not very straight forward and I was lazy here.

Comment (brizental) [reply glean/tests/unit/core/glean.spec.ts:438]: Lazyness fixed by [5c9e135](https://github.com/mozilla/glean.js/pull/1228/commits/5c9e135c81575fe2004466994c210e620da43d89) :D Thanks for calling this out

**Pull request merged**